### PR TITLE
LIME-1129 - Infra Tests - Alarms for HMRC KBV

### PR DESCRIPTION
**Pull request by Z-Mohamoud**
### What changed

Add Infra tests to validate Alarms config in CloudFormation template. 

All tests passing locally:
image

### Issue tracking

- [LIME-1129](https://govukverify.atlassian.net/browse/LIME-1129)

### Other considerations

[LIME-1129]: https://govukverify.atlassian.net/browse/LIME-1129?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: integration-tests/jest.config.ts [hunk 1]
```diff
@@ -3,8 +3,6 @@ import baseConfig from "../jest.config.base";
 
 export default {
   ...baseConfig,
-  projects: ["tests/*/jest.config.ts"],
-  testMatch: ["<rootDir>/**/*.test.ts"],
-  collectCoverage: false,
+  testMatch: ["<rootDir>/tests/infra/**/*.test.ts"], // Adjust to specifically match infra tests
```
Comment (gds-jbentham) [inline integration-tests/jest.config.ts:6]: Do we not run any other tests from the integration-test folder at the moment?

File: infrastructure/template.yaml [hunk 2]
```diff
@@ -250,7 +250,7 @@ Resources:
   PublicKbvHmrcApiAccessLogGroup:
     Type: AWS::Logs::LogGroup
     Properties:
-      LogGroupName: !Sub /aws/vendedlogs/apigateway/${AWS::StackName}-${PublicKbvHmrcApi}-public-AccessLogs
+      LogGroupName: !Sub /aws/vendedlogs/apigateway/${AWS::StackName}-public-AccessLogs
```
Comment (gds-jbentham) [inline infrastructure/template.yaml:253]: How come we are changing these names?

Comment (ChrisBates1) [reply infrastructure/template.yaml:253]: Hi @gds-jbentham this was raised by the script as a issue, @Z-Mohamoud has detailed the issue in the comments on [LIME-1258](https://govukverify.atlassian.net/browse/LIME-1258)

File: integration-tests/jest.config.ts (hunk 1 above)
Comment (Z-Mohamoud) [reply integration-tests/jest.config.ts]: Apologies, I removed these when debugging issues with the jest.config. I've reverted these changes now

File: infrastructure/template.yaml (hunk 2 above)
Comment (gds-jbentham) [reply infrastructure/template.yaml:253]: @smsgds could you have a look at this

Comment (smsgds) [reply infrastructure/template.yaml:253]: We had a look at this, there is no circular dependancies due to the use of getAttr and Ref. 

- The api is always created first
- Then the loggroup, log group sub filters and api stage are all created next.

So the names can be reverted back

**Review by smsgds — CHANGES_REQUESTED** [review]

File: infrastructure/template.yaml (hunk 2 above)
Comment (smsgds) [reply infrastructure/template.yaml:247]: Approving as is, the aws-cdk-lib used is unaware of the api gateway being created separately from the stage. Leading to it thinking the gateway name would be unavailable.

The same naming change will probably be needed when template checks are added to fraud, dl and passport.

**Review by smsgds — APPROVED** [review]

**Review by smsgds — APPROVED** [review]
https://github.com/govuk-one-login/ipv-cri-kbv-hmrc-api/pull/116/files#r1754117025

**Pull request merged**